solidity/contracts/vault/TBTCOptimisticMinting.sol

```diff
@@ -0,0 +1,375 @@
+            depositKey
+        );
+    }
+
+    /// @notice Adds the address to the Minter set.
+    function addMinter(address minter) external onlyOwner {
+        require(!isMinter[minter], "This address is already a minter");
+        isMinter[minter] = true;
+        emit MinterAdded(minter);
+    }
+
+    /// @notice Removes the address from the Minter set.
+    function removeMinter(address minter) external onlyOwnerOrGuardian {
+        require(isMinter[minter], "This address is not a minter");
+        delete isMinter[minter];
+        emit MinterRemoved(minter);
+    }
+
+    /// @notice Adds the address to the Guardian set.
+    function addGuardian(address guardian) external onlyOwner {
+        require(!isGuardian[guardian], "This address is already a guardian");
+        isGuardian[guardian] = true;
+        emit GuardianAdded(guardian);
+    }
+
+    /// @notice Removes the address from the Guardian set.
+    function removeGuardian(address guardian) external onlyOwner {
+        require(isGuardian[guardian], "This address is not a guardian");
+        delete isGuardian[guardian];
+        emit GuardianRemoved(guardian);
+    }
+
+    /// @notice Pauses the optimistic minting. Note that the pause of the
+    ///         optimistic minting does not stop the standard minting flow
+    ///         where wallets sweep deposits.
+    function pauseOptimisticMinting() external onlyOwner {
+        require(
+            !isOptimisticMintingPaused,
+            "Optimistic minting already paused"
+        );
+        isOptimisticMintingPaused = true;
+        emit OptimisticMintingPaused();
+    }
+
+    /// @notice Unpauses the optimistic minting.
+    function unpauseOptimisticMinting() external onlyOwner {
+        require(isOptimisticMintingPaused, "Optimistic minting is not paused");
+        isOptimisticMintingPaused = false;
+        emit OptimisticMintingUnpaused();
+    }
+
+    /// @notice Calculates deposit key the same way as the Bridge contract.
+    ///         The deposit key is computed as
+    ///         keccak256(fundingTxHash | fundingOutputIndex).
+    function calculateDepositKey(
+        bytes32 fundingTxHash,
+        uint32 fundingOutputIndex
+    ) public view returns (uint256) {
+        return
+            uint256(
+                keccak256(abi.encodePacked(fundingTxHash, fundingOutputIndex))
+            );
+    }
+
+    /// @notice Used by TBTCVault.receiveBalanceIncrease to repay the optimistic
+    ///         minting debt before TBTC is minted. When optimistic minting is
+    ///         finalized, debt equal to the value of the deposit being
+    ///         a subject of the optimistic minting is incurred. When TBTCVault
+    ///         sweeps a deposit, the debt is fully or partially paid off, no
+    ///         matter if that particular deposit was used for the optimistic
+    ///         minting or not.
+    function repayOptimisticMintDebt(address depositor, uint256 amount)
```

Comment:
Could we describe in doc what is this function expected to return and what exactly is the `amount` parameter?